integration/src/integration/java/pl/allegro/tech/hermes/integration/BatchDeliveryTest.java

```diff
@@ -0,0 +1,132 @@
+package pl.allegro.tech.hermes.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.integration.env.SharedServices;
+import pl.allegro.tech.hermes.test.helper.avro.AvroUser;
+import pl.allegro.tech.hermes.test.helper.endpoint.RemoteServiceEndpoint;
+import pl.allegro.tech.hermes.test.helper.message.TestMessage;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.stream;
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+import static pl.allegro.tech.hermes.api.Topic.Builder.topic;
+import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
+
+public class BatchDeliveryTest extends IntegrationTest {
+    private static final int LARGE_MESSAGE_TTL = 100 * 1000;
+    private RemoteServiceEndpoint remoteService;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeMethod
+    public void initializeAlways() {
+        this.remoteService = new RemoteServiceEndpoint(SharedServices.services().serviceMock());
+    }
+
+    private static final int LARGE_BATCH_TIME = Integer.MAX_VALUE;
+    private static final int LARGE_BATCH_SIZE = 100;
+    private static final int LARGE_BATCH_VOLUME = 1024;
+
+    private static final int SMALL_BATCH_TIME = 1;
+    private static final int SMALL_BATCH_SIZE = 2;
+    private static final TestMessage[] SMALL_BATCH = TestMessage.simpleMessages(SMALL_BATCH_SIZE);
+
+    private static final TestMessage SINGLE_MESSAGE = TestMessage.simple();
+
+    @Test
+    public void shouldDeliverMessagesInBatch() throws IOException {
+        // given
+        Topic topic = operations.buildTopic("batchSizeTest", "topic");
+        operations.createBatchSubscription(topic, HTTP_ENDPOINT_URL, LARGE_MESSAGE_TTL, SMALL_BATCH_SIZE, LARGE_BATCH_TIME, LARGE_BATCH_VOLUME);
+        remoteService.expectMessages(SMALL_BATCH);
+
+        // when
+        stream(SMALL_BATCH).forEach(m -> publish(topic, m));
+
+        // then
+        expectSingleBatch(SMALL_BATCH);
+    }
+
+    @Test
+    public void shouldDeliverBatchInGivenTimePeriod() throws IOException {
+        // given
+        Topic topic = operations.buildTopic("deliverBatchInGivenTimePeriod", "topic");
+        operations.createBatchSubscription(topic, HTTP_ENDPOINT_URL, LARGE_MESSAGE_TTL, LARGE_BATCH_SIZE, SMALL_BATCH_TIME, LARGE_BATCH_VOLUME);
```

Comment:
repeating the same BIG_VALUES doesnt make it more readable.. either make a builder for batch subscription or aply some defaults, or just use values instead of consts